### PR TITLE
Fix warnings with llvm10 and -Wrange-loop-construct

### PR DESCRIPTION
**Pull request by krionbsd**
### Short description
Fix warnings with llvm10 and -Wrange-loop-construct

### Checklist
I have:
- [x] compiled this code
- [x] tested this code

### CODE REVIEW

**Comment by omoerbeek** [comment]
Can you show an example warning for one of the `const &` cases? I don't have clang10 at hand.

**Comment by krionbsd** [comment]
https://krion.cc/data/head-amd64-default/2020-04-05_20h19m34s/logs/powerdns-recursor-4.3.0.log

**Comment by krionbsd** [comment]
Here is the build with fixed warnings:
https://krion.cc/data/head-amd64-default/2020-04-06_09h31m24s/logs/powerdns-recursor-4.3.0.log

I deliberately didn't fix deprecated things for now

**Comment by omoerbeek** [comment]
Nice, thanks.

File: pdns/dnssecinfra.cc [hunk 1]
```diff
@@ -137,7 +137,7 @@ std::string DNSCryptoKeyEngine::convertToISC() const
   storvector_t stormap = this->convertToISCVector();
```
Comment (omoerbeek) [inline pdns/dnssecinfra.cc:137]:
```suggestion
  storvector_t storvector = this->convertToISCVector();
```

File: pdns/dnssecinfra.cc [hunk 2]
```diff
@@ -137,7 +137,7 @@ std::string DNSCryptoKeyEngine::convertToISC() const
...
   ostringstream ret;
   ret<<"Private-key-format: v1.2\n";
-  for(const stormap_t::value_type& value :  stormap) {
+  for(const storvector_t::value_type& value :  stormap) {
```
Comment (omoerbeek) [inline pdns/dnssecinfra.cc:140]:
```suggestion
  for(const storvector_t::value_type& value :  storvec) {
```

Comment (omoerbeek) [reply pdns/dnssecinfra.cc:140]: IMO this clears up the confusion...

Comment (krionbsd) [reply pdns/dnssecinfra.cc:140]: Yes, agree, I think `for(const auto& value :  stormap) {` would work as well but I've to test it first

**Comment by krionbsd** [comment]
Latest build looks good: https://krion.cc/data/head-amd64-default/2020-04-06_11h24m45s/logs/powerdns-recursor-4.3.0.log

**Pull request merged**